Adding Functional side layout

will-jac opened the pull request:
This PR does 2 things:

- Moves all the layout code to it's own directory
- Finishes the half-implemented side layout

The side layout is mobile oriented. If the expand / collapse charms were spaced a bit further and somewhat simplified, it would likely be a highly usable iPad layout (testing has not yet been completed due to current lack of an iPad).

The layout features a vertical tab bar on the left for switching between the Editor/Console, Robot, and World menus. The current tab can be selected to minimize the sidebar (effectively maximizing the simulator).

Layout: 
![image](https://user-images.githubusercontent.com/32182937/162876412-545e965b-bf7a-42a5-b554-f94263bda698.png)

bmcdorman commented on src/components/TabBar.tsx:
```diff
@@ -7,42 +7,50 @@ import { ThemeProps } from './theme';
+  position: 'absolute', 
+  left: '50%', 
+  top: '50%',
+  transform: (props.$vertical) ? 'translate(-50%, -50%) rotate(-90deg)' : null,
```
The `translate` should apply whether or not the text is vertical.

will-jac replied:
Should be fixed now, good call

will-jac commented:
Some updates:

- I added vertical sliders (this is easy to roll-back if desired)
- This PR fixes #345, which was required for the vertical slider to work properly
  - Fixed via using `window.inner-height` to calculate the heights instead of `100vh`
- Layout and UI is generally a bit more thought-through
  - click the tabbar to collapse / expand the side bar, slide to adjust width
